src/windows/list.cpp

```diff
@@ -0,0 +1,687 @@
+0x53F56311L, 0xB6BF, 0x11D0, { 0x94, 0xF2, 0x00, 0xA0, 0xC9, 0x1E, 0xFB, 0x8B }
+};
+
+const GUID GUID_DEVICE_INTERFACE_WRITEONCEDISK = {
+0x53F5630CL, 0xB6BF, 0x11D0, { 0x94, 0xF2, 0x00, 0xA0, 0xC9, 0x1E, 0xFB, 0x8B }
+};
+
+const GUID GUID_DEVICE_INTERFACE_TAPE = {
+0x53F5630BL, 0xB6BF, 0x11D0, { 0x94, 0xF2, 0x00, 0xA0, 0xC9, 0x1E, 0xFB, 0x8B }
+};
+
+const GUID GUID_DEVICE_INTERFACE_USB_DEVICE = {
+0xA5DCBF10L, 0x6530, 0x11D2, { 0x90, 0x1F, 0x00, 0xC0, 0x4F, 0xB9, 0x51, 0xED }
+};
+
+const GUID GUID_DEVICE_INTERFACE_VOLUME = {
+0x53F5630DL, 0xB6BF, 0x11D0, { 0x94, 0xF2, 0x00, 0xA0, 0xC9, 0x1E, 0xFB, 0x8B }
+};
+
+const GUID GUID_DEVICE_INTERFACE_STORAGEPORT = {
+0x2ACCFE60L, 0xC130, 0x11D2, { 0xB0, 0x82, 0x00, 0xA0, 0xC9, 0x1E, 0xFB, 0x8B }
+};
+
+const std::vector<std::string> USB_STORAGE_DRIVERS {
+  "USBSTOR", "UASPSTOR", "VUSBSTOR",
+  "RTUSER", "CMIUCR", "EUCR",
+  "ETRONSTOR", "ASUSSTPT"
+};
+
+const std::vector<std::string> GENERIC_STORAGE_DRIVERS {
+  "SCSI", "SD", "PCISTOR",
+  "RTSOR", "JMCR", "JMCF", "RIMMPTSK", "RIMSPTSK", "RIXDPTSK",
+  "TI21SONY", "ESD7SK", "ESM7SK", "O2MD", "O2SD", "VIACR"
+};
+
+// List of known virtual disk hardware IDs
+const std::vector<std::string> VHD_HARDWARE_IDS {
+  "Arsenal_________Virtual_",
+  "KernSafeVirtual_________",
+  "Msft____Virtual_Disk____",
+  "VMware__VMware_Virtual_S"
+};
+
+char* WCharToUtf8(const wchar_t* wstr) {
+  char *str = NULL;
+  size_t size = WideCharToMultiByte(CP_UTF8, 0, wstr, -1, NULL, 0, NULL, NULL);
+
+  if (size <= 1) {
+    return NULL;
+  }
+
+  if ((str = reinterpret_cast<char*>(calloc(size, 1))) == NULL) {
+    return NULL;
+  }
+
+  size_t utf8Size = WideCharToMultiByte(
+    CP_UTF8, 0, wstr, -1, str, size, NULL, NULL);
+
+  if (utf8Size != size) {
+    free(str);
+    return NULL;
+  }
+
+  return str;
+}
+
+char* GetEnumeratorName(HDEVINFO hDeviceInfo, SP_DEVINFO_DATA deviceInfoData) {
+  DWORD size;
+  DWORD dataType;
+  char buffer[MAX_PATH];
+
+  ZeroMemory(&buffer, sizeof(buffer));
+
+  BOOL hasEnumeratorName = SetupDiGetDeviceRegistryPropertyA(
+    hDeviceInfo, &deviceInfoData, SPDRP_ENUMERATOR_NAME,
+    &dataType, (LPBYTE) buffer, sizeof(buffer), &size);
+
+  return hasEnumeratorName ? buffer : NULL;
+}
+
+std::string GetFriendlyName(HDEVINFO hDeviceInfo,
+  SP_DEVINFO_DATA deviceInfoData) {
+  DWORD size;
+  DWORD dataType;
+  wchar_t wbuffer[MAX_PATH];
+
+  ZeroMemory(&wbuffer, sizeof(wbuffer));
+
+  BOOL hasFriendlyName = SetupDiGetDeviceRegistryPropertyW(
+    hDeviceInfo, &deviceInfoData, SPDRP_FRIENDLYNAME,
+    &dataType, (PBYTE) wbuffer, sizeof(wbuffer), &size);
+
+  return hasFriendlyName ? WCharToUtf8(wbuffer) : std::string("");
+}
+
+bool IsSCSIDevice(std::string enumeratorName) {
+  for (std::string driverName : GENERIC_STORAGE_DRIVERS) {
+    if (enumeratorName == driverName) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+bool IsUSBDevice(std::string enumeratorName) {
+  for (std::string driverName : USB_STORAGE_DRIVERS) {
```

Comment:
Same here. I think we should use `std::set`

Reply:
And done!